engine/resources/mount.go

```diff
@@ -0,0 +1,712 @@
+// Mgmt
+// Copyright (C) 2013-2018+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/purpleidea/mgmt/engine"
+	"github.com/purpleidea/mgmt/engine/traits"
+	"github.com/purpleidea/mgmt/recwatch"
+	"github.com/purpleidea/mgmt/util"
+
+	sdbus "github.com/coreos/go-systemd/dbus"
+	systemdUtil "github.com/coreos/go-systemd/util"
+	fstab "github.com/deniswernert/go-fstab"
+	"github.com/godbus/dbus"
+	errwrap "github.com/pkg/errors"
+)
+
+func init() {
+	engine.RegisterResource("mount", func() engine.Res { return &MountRes{} })
+}
+
+const (
+	// procPath is the path to /proc/mounts which contains all active mounts.
+	procPath = "/proc/mounts"
+	// fstabPath is the path to the fstab file which defines mounts.
+	fstabPath = "/etc/fstab"
+	// fstabUmask is the umask (permissions) used to edit /etc/fstab.
+	fstabUmask = 0644
+
+	// getStatus64 is an ioctl command to get the status of file backed
+	// loopback devices (i.e. iso file mounts.)
+	getStatus64 = 0x4C05
+	// loopFileUmask is the umask (permissions) used to read the loop file.
+	loopFileUmask = 0660
+
+	// devDisk is the path where disks and partitions can be found, organized
+	// by uuid/label/path.
+	devDisk = "/dev/disk/"
+	// diskByUUID is the location of symlinks for devices by UUID.
+	diskByUUID = devDisk + "by-uuid/"
+	// diskByLabel is the location of symlinks for devices by label.
+	diskByLabel = devDisk + "by-label/"
+	// diskByUUID is the location of symlinks for partitions by UUID.
+	diskByPartUUID = devDisk + "by-partuuid/"
+	// diskByLabel is the location of symlinks for partitions by label.
+	diskByPartLabel = devDisk + "by-partlabel/"
+
+	// dbusStoragedPath is the dbus path that watch eavesdrops on for signals.
+	dbusStoragedPath = "/org/freedesktop/UDisks2"
+)
+
+// vfsTypes is a list of all valid filesystem types taken from
+// /proc/filesystems.
+var vfsTypes = []string{
```

Comment:
Are these in a particular order? If not let's make it alphabetical please. Mention whatever order is being used in the docstring.

Reply:
vim :sort to the rescue!

Reply:
You're right. I was going to figure that out, and then I forgot. :/